### PR TITLE
[SPARK-44681] Fix issues when writing Go application code using Spark Connect Go client library

### PR DESCRIPTION
**Pull request by hiboyang**
### What changes were proposed in this pull request?

When trying to write Go application code using Spark Connect Go client and put the Go application code in users' own repo (e.g. https://github.com/user-foo/my-go-application), the Go application code cannot resolve Spark Connect Go client from `https://github.com/apache/spark-connect-go` correctly, due to two issues:

1. The name `github.com/apache/spark-connect-go/v_3_4` cannot resolve correctly to find the module, it complains not finding `go.mod` file under `github.com/apache/spark-connect-go/v_3_4`. After change the name to `github.com/apache/spark-connect-go/v34`, it is good.

2. The Go application code needs generated Go protobuf code as well. Thus need to commit the generated Go protobuf code into `github.com/apache/spark-connect-go` repo.

### Why are the changes needed?

See above.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Tested by run Go application code to use Spark Connect Go client.

### CODE REVIEW

**Comment by hiboyang** [comment]
@HyukjinKwon @grundprinzip @zhengruifeng I find some issues when I write a standalone Spark Connect application using this spark-connect-go repo as a library. The PR is to fix these issues.

**Comment by hiboyang** [comment]
This PR is mostly generated code like `internal/generated/base.pb.go`, `internal/generated/base_grpc.pb.go`, etc. We need to put those files into this repo so other Go application could reference this repo as a Go library properly.

Please ignore those generated code. Except those code, the PR is actually small :)

**Review by grundprinzip — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -13,7 +13,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-module github.com/apache/spark-connect-go/v_3_4
+module github.com/apache/spark-connect-go/v34
```
[the remark this answers is not in the document]
Comment (hiboyang) [reply go.mod:16]: Thanks for reviewing! Yeah, previously we reference the code inside this repo, thus `v_3_4` does not cause problem. When I wrote another Go application outside of this repo, referencing this repo as a library, then got issue.

**Comment by zhengruifeng** [comment]
merged to master

**Pull request closed without merging**